frontend/packages/ux-editor/src/components/Properties/PropertiesHeader/EditLayoutSetForSubform/EditLayoutSet/CreateNewSubformSection/SubformDataModel.tsx

```diff
@@ -1,31 +1,47 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { StudioTextfield, StudioNativeSelect, StudioProperty } from '@studio/components';
 import { LinkIcon } from '@studio/icons';
 import { useTranslation } from 'react-i18next';
 import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
 import { useAppMetadataModelIdsQuery } from 'app-shared/hooks/queries/useAppMetadataModelIdsQuery';
 import classes from './SubformDataModel.module.css';
+import { useValidateSchemaName } from 'app-shared/hooks/useValidateSchemaName';
+import { extractDataTypeNamesFromAppMetadata } from 'app-development/features/dataModelling/SchemaEditorWithToolbar/TopToolbar/utils/validationUtils';
+import { useAppMetadataQuery } from 'app-shared/hooks/queries';
 
 export type SubformDataModelProps = {
   setDisplayDataModelInput: (setDisplayDataModelInput: boolean) => void;
   setNewDataModel: (dataModelId: string) => void;
   displayDataModelInput: boolean;
   setSelectedDataModel: (dataModelId: string) => void;
+  setDataModelError?: (error: string | undefined) => void;
 };
 
 export const SubformDataModel = ({
   setDisplayDataModelInput,
   setSelectedDataModel,
   setNewDataModel,
   displayDataModelInput,
+  setDataModelError,
 }: SubformDataModelProps): React.ReactElement => {
   const { t } = useTranslation();
   const { org, app } = useStudioEnvironmentParams();
   const { data: dataModelIds } = useAppMetadataModelIdsQuery(org, app, false);
 
+  const { data: appMetadata } = useAppMetadataQuery(org, app);
+  const dataTypeNames = extractDataTypeNamesFromAppMetadata(appMetadata);
+  const { validateName, nameError: dataModelNameError } = useValidateSchemaName(
+    dataModelIds,
+    dataTypeNames,
+  );
+
+  useEffect(() => {
+    setDataModelError(dataModelNameError);
+  }, [dataModelNameError, setDataModelError]);
+
   const handleDataModel = (dataModelId: string) => {
-    // TODO: https://github.com/Altinn/altinn-studio/issues/14184
-    setNewDataModel(dataModelId);
+    validateName(dataModelId);
+    if (!dataModelNameError) setNewDataModel(dataModelId);
```

Comment:
The `newDataModel` state was intended as a temporary solution for simple validation of a new data model, but I don't think it's needed anymore now that we have the validation hook. Since the content is wrapped in a form with submit in the `nextRecommendedAction` component, we don't need to save the state unless it's specifically for validation.

See the variable `hasInvalidSubformName` that has a similar approach with a validation hook.

Reply:
fixed